webapp/src/selectors/index.ts

```diff
@@ -13,17 +13,14 @@ const getPluginState = (state) => state['plugins-' + PluginId] || {};
 
 export const getPluginServerRoute = (state) => {
     const config = getConfig(state);
+    const baseUrl = new URL((config && config.SiteURL) || window.location.href);
+    let basePath = baseUrl.pathname;
 
-    let basePath = '';
-    if (config && config.SiteURL) {
-        basePath = new URL(config.SiteURL).pathname;
-
-        if (basePath && basePath[basePath.length - 1] === '/') {
-            basePath = basePath.substr(0, basePath.length - 1);
-        }
+    if (basePath && basePath[basePath.length - 1] === '/') {
+        basePath = basePath.substr(0, basePath.length - 1);
```

Comment:
If we use `window.location.href`, we'll just want to take the `origin` (as we are below), instead of including the `pathname`. If we take `pathname` from `window.location.href`, that will result in a value like `/core/messages/@dylan.haussermann` in the case of `https://community.mattermost.com/core/messages/@dylan.haussermann`.

Subpaths are only supported through the `SiteURL`, so if the `SiteURL` is not set, we can expect the server to not be using a subpath. The code that was here before seems correct.

Reply:
@mickmister reverted this change